frontends/main/src/page-components/EnrollmentDialogs/CourseEnrollmentDialog.test.tsx

```diff
@@ -0,0 +1,391 @@
+import {
+  urls as mitxUrls,
+  factories as mitxFactories,
+} from "api/mitxonline-test-utils"
+import type { CourseWithCourseRunsSerializerV2 } from "@mitodl/mitxonline-api-axios/v2"
+import NiceModal from "@ebay/nice-modal-react"
+import CourseEnrollmentDialog from "./CourseEnrollmentDialog"
+import { upgradeRunUrl } from "@/common/mitxonline"
+import { faker } from "@faker-js/faker/locale/en"
+import invariant from "tiny-invariant"
+import { DASHBOARD_HOME } from "@/common/urls"
+
+const makeCourseRun = mitxFactories.courses.courseRun
+const makeProduct = mitxFactories.courses.product
+const makeCourse = mitxFactories.courses.course
+
+const enrollableRun: typeof makeCourseRun = (overrides) =>
+  makeCourseRun({
+    is_enrollable: true,
+    enrollment_start: faker.date.past().toISOString(),
+    enrollment_end: faker.date.future().toISOString(),
+    ...overrides,
+  })
+
+const upgradeableRun: typeof makeCourseRun = (overrides) =>
+  makeCourseRun({
+    is_upgradable: true,
+    is_enrollable: true,
+    is_archived: false,
+    products: [mitxFactories.courses.product()],
+    ...overrides,
+  })
+
+describe("CourseEnrollmentDialog", () => {
+  const openDialog = async (course: CourseWithCourseRunsSerializerV2) => {
+    await act(async () => {
+      NiceModal.show(CourseEnrollmentDialog, { course })
+    })
+    return await screen.findByRole("dialog")
+  }
+
+  setupLocationMock()
+
+  describe("Course run dropdown", () => {
+    test("Shows one entry for each enrollable course run", async () => {
+      const run1 = enrollableRun()
+      const run2 = enrollableRun()
+      const run3 = enrollableRun()
+      const course = makeCourse({ courseruns: [run1, run2, run3] })
+
+      renderWithProviders(null)
+      await openDialog(course)
+
+      const select = screen.getByRole("combobox", { name: /choose a date/i })
+      await user.click(select)
+
+      const options = await screen.findAllByRole("option")
+      // Should have 4 options: 1 "Please Select" + 3 course runs
+      expect(options).toHaveLength(4)
+
+      // Verify the actual run options (excluding "Please Select")
+      const runOptions = options.slice(1)
+      expect(runOptions).toHaveLength(3)
+    })
+
+    test("Does NOT include non-enrollable course runs in dropdown", async () => {
+      const run1 = enrollableRun()
+      const nonEnrollableRun = makeCourseRun({ is_enrollable: false })
+      const archivedRun = makeCourseRun({
+        is_archived: true,
+        is_enrollable: true,
+      })
+      const course = makeCourse({
+        courseruns: [run1, nonEnrollableRun, archivedRun],
+      })
+
+      renderWithProviders(<div />)
+      await openDialog(course)
+
+      const select = screen.getByRole("combobox", { name: /choose a date/i })
+      await user.click(select)
+
+      const options = await screen.findAllByRole("option")
+      expect(options).toHaveLength(3) // 1 "Please Select" + 2 enrollable runs
```

Comment:
Above you slice out the "Please Select" option. Not really a big deal, but should we do that here?